client/data/settings/selectors.js

```diff
@@ -0,0 +1,28 @@
+/** @format */
+
+const EMPTY_OBJ = {};
+const EMPTY_ARR = [];
+
+const getSettingsState = ( state ) => {
+	if ( ! state ) {
+		return EMPTY_OBJ;
+	}
+
+	return state.settings || EMPTY_OBJ;
+};
+
+export const getSettings = ( state ) => {
+	return getSettingsState( state ).data || EMPTY_OBJ;
+};
+
+export const isSavingSettings = ( state ) => {
+	return getSettingsState( state ).isSaving || false;
+};
+
+export const getIsPaymentRequestEnabled = ( state ) => {
+	return getSettings( state ).is_payment_request_enabled || false;
+};
```

Comment:
I understand that we used the same approach on WCPay - however: :D 

Most of these selectors are only used in one instance in the codebase (in the hooks).

What are your thoughts on mostly leveraging the `getSettings()` selector in the hooks?
This would shave some weight on the bundle size (since we wouldn't need to bundle these additional functions (that cannot be minified) and we could maintain a little less code).

So in the hooks, we could just do (as an example):
```js
export const usePaymentRequestEnabledSettings = () => {
	const { updateIsPaymentRequestEnabled } = useDispatch( STORE_NAME );

	const isPaymentRequestEnabled = useSelect( ( select ) => {
		const { getSettings } = select( STORE_NAME );

		return getSettings().is_payment_request_enabled || false;
	}, [] );

	return [ isPaymentRequestEnabled, updateIsPaymentRequestEnabled ];
};
```

Reply:
Yes! This will be a departure from what we do in WCPay but as you said, the only client code for these functions are the hooks.